qiskit/transpiler/passes/synthesis/unitary_synthesis.py

```diff
@@ -331,6 +331,14 @@ def run(self, dag: DAGCircuit) -> DAGCircuit:
                     dag.substitute_node_with_dag(node, synth_dag[0], wires=synth_dag[1])
                 else:
                     dag.substitute_node_with_dag(node, synth_dag)
+        for node in dag.control_flow_ops():
+            updated_blocks = []
+            for block in node.op.blocks:
+                dag_block = circuit_to_dag(block)
+                updated_dag_block = self.run(dag_block)
+                updated_circ_block = dag_to_circuit(updated_dag_block)
+                updated_blocks.append(updated_circ_block)
+            node.op = node.op.replace_blocks(updated_blocks)
```

Comment:
Did we have a standard method of being lazy here if we didn't need to update the blocks?

Reply:
I can apply a check

Reply:
Sorry, in retrospect, there's no need - I'd misremembered what we'd done elsewhere for this first pass.  I think the check will be more appropriate once we've got a better structure for representing control-flow within the transpiler.  Doing it now is just busy work for something we're going to throw out again.

(I know we already did it in the basis translator - sorry about that.)